.github/copilot-instructions.md

```diff
@@ -0,0 +1,267 @@
+
+**JavaScript Assets:** Located in each gem's `app/packs/` directory
+**Stylesheets:** Located in each gem's `app/packs/stylesheets/` directory
+
+### Common Development Tasks
+
+**Running the development server:**
+```bash
+cd spec/decidim_dummy_app
+bin/dev  # Starts Rails + webpack dev server
+# Access at http://localhost:3000
+# Admin panel: http://localhost:3000/admin (after creating admin user)
+```
+
+**Database operations:**
+```bash
+cd spec/decidim_dummy_app
+bin/rails db:create     # Create database
+bin/rails db:migrate    # Run migrations  
+bin/rails db:seed       # Load sample data
+bin/rails db:reset      # Reset and reseed database
+```
+
+**Asset compilation:**
+```bash
+cd spec/decidim_dummy_app
+bin/rails assets:precompile
+```
+
+## Validation Scenarios
+
+**CRITICAL**: Always test actual functionality after making changes:
+
+### User Registration and Login Flow
+```bash
+cd spec/decidim_dummy_app
+bin/dev
+# 1. Visit http://localhost:3000
+# 2. Click "Sign up"
+# 3. Create a new user account
+# 4. Verify email confirmation process works
+# 5. Log in with the new account
+# 6. Navigate to user profile and edit settings
+```
+
+### Admin Panel Access
+```bash
+cd spec/decidim_dummy_app
+bin/rails console
+# Create admin user:
+user = Decidim::User.create!(
+  email: "admin@example.org",
+  password: "decidim123456",
+  name: "System Admin",
+  organization: Decidim::Organization.first,
+  confirmed_at: Time.current,
+  admin: true
+)
+# Visit http://localhost:3000/admin with these credentials
+```
+
+### Creating a Participatory Process
+1. Log in as admin
+2. Go to Admin Panel > Participatory Processes  
+3. Create new process with basic information
+4. Add components (proposals, meetings, surveys)
+5. Publish the process
+6. Test public access and participation features
+
+## Build Pipeline Integration
+
+**CI Requirements:** The `.github/workflows/` contain the production CI setup:
+- Tests run on Ubuntu 22.04
+- Requires PostgreSQL and Redis services
+- Uses specific Ruby and Node versions
+- Runs parallel tests across multiple modules
+- **Each module CI can timeout at 30-60 minutes - NEVER CANCEL**
+
+**Before committing, ALWAYS run:**
+```bash
+npm run lint      # JavaScript linting
+bundle exec rubocop  # Ruby linting  
+bundle exec rspec    # Main test suite - NEVER CANCEL (1m49s)
+npm run test      # JavaScript tests (13s)
+```
+
+## Troubleshooting Common Issues
+
+**Bundle install fails:** Usually Ruby version mismatch. Must use Ruby 3.3.4.
+
+**Test app creation fails:** Check Redis and PostgreSQL are running with correct credentials.
+
+**Asset compilation errors:** Ensure Node 22.14.0 is active and npm dependencies installed.
+
+**Permission errors during gem installs:** Ensure rbenv is properly installed and configured. Run `rbenv rehash` after installing new gems.
+
+**Database connection errors:** Set `DATABASE_USERNAME` and `DATABASE_PASSWORD` environment variables.
+
+## Important Notes
```

Comment:
Another "Important Note": some of the changes done in `decidim-generator` that apply to the application configuration (files like `config/application.rb`, `config/environments/*`, etc) or other files generated through the generator, should be also informed through "Releases Notes" (in `RELEASE_NOTES.md`). See https://github.com/decidim/decidim/pull/15192#pullrequestreview-3233794349

Reply:
@andreslucena I would also run automatically the `rake bundle` to update the generator gemfile when needed ? like "if any .gemspec file is changed, or if Gemfile.lock is changed, automatically run ..."?